src/JsonSchema/Constraints/CollectionConstraint.php

```diff
@@ -111,4 +130,17 @@ protected function validateItems($value, $schema = null, JsonPointer $path = nul
             }
         }
     }
+
+    const OPTIMIZED_ITEM_TYPES = array('string', 'number', 'integer');
```

Comment:
You'll have to come up with some other way to encode these values, unfortunatley; older versions of PHP don't allow assigning arrays to constants.